### ISSUE TITLE
Move kubearmor.io to new static generator

### ISSUE THREAD

**Issue by ShubhamTatvamasi** [kubearmor/kubearmor.io#1]
Currently, our [website](https://github.com/accuknox/KubeArmor-website/) is on build using gatsby. We are planning to move it to a new static website generator, so we can host it on github pages. I created a draft using [hugo](https://github.com/ShubhamTatvamasi/hugo-demo) for our blogging needs. We can look for more options if needed.

**Comment by Aditya-150** [comment]
I have an idea. We can directly use GitHub wiki page content for the blog for this we can use GitHub API.
What say? @ShubhamTatvamasi

**Comment by ShubhamTatvamasi** [comment]
We are doing that already. We want people to create a PR for posting new blogs and have a custom theme for our landing page.
https://github.com/kubearmor/KubeArmor/wiki

**Comment by Aditya-150** [comment]
I understand it now.

> We are doing that already. We want people to create a PR for posting new blogs and have a custom theme for our landing page. https://github.com/kubearmor/KubeArmor/wiki

**Comment by Aditya-150** [comment]
When the PR having tag of blog is merged. It should be released in the blogs section of the website. I think we do it by setting up a github workflow for it.

**Comment by Aditya-150** [comment]
> We are doing that already. We want people to create a PR for posting new blogs and have a custom theme for our landing page. https://github.com/kubearmor/KubeArmor/wiki

Also theme and UI for website. If you want I can design.

**Comment by ShubhamTatvamasi** [comment]
I worked on this theme, can we improve it as per our needs?
https://hugo.shubhamtatvamasi.com

**Comment by Aditya-150** [comment]
Yeah, we can definitely improve the theme. Even we can make or revamp whole website of [kuberarmor.io ](https://kuberarmor.io/) using [hugo](https://gohugo.io/) unless its using any dynamic functionalities. 

As I can see you are using theme provided by [hugo-PaperMod](https://github.com/adityatelange/hugo-PaperMod). We need to add our own stylesheets, CSS files or CSS modules in order to customize the theme of the website.

_**My question is: Will we be revamping the whole [kuberarmor.io ] or the blog part you are currently working on?**_

**Comment by ShubhamTatvamasi** [comment]
We will have to change the whole website. Come up with new design.

**Comment by Aditya-150** [comment]
Fine👍!  Should I create a Figma file and share the access of it via mail(mentioned in your profile) ? or we can do the vice-versa.

**Comment by ShubhamTatvamasi** [comment]
You do it.

**Comment by Aditya-150** [comment]
Done🚀

**Comment by ShubhamTatvamasi** [comment]
I got the email, we have to get on a call for this.

**Comment by Aditya-150** [comment]
Yup. Anytime in night. I'll be available and in any platform you choose.

**Comment by Aditya-150** [comment]
> Yup. Anytime in night. I'll be available and in any platform you choose.

Just notify me before hand if possible. I'll be there.

**Comment by Aditya-150** [comment]
@ShubhamTatvamasi  I'm just thinking of conducting an user survey in order to get an idea of what people think about current design and what changes or improvements they want in order to make the site more intuitive. Do you think this would work?

Also a new channel can only be create by the admin of the slack workspace. We need to talk to Rahul about this as he currently has the admin access.

**Comment by ShubhamTatvamasi** [comment]
> @ShubhamTatvamasi I'm just thinking of conducting an user survey in order to get an idea of what people think about current design and what changes or improvements they want in order to make the site more intuitive. Do you think this would work?
> 
> Also a new channel can only be create by the admin of the slack workspace. We need to talk to Rahul about this as he currently has the admin access.

@nyrahul can you please create a new channel for **#kuberarmor-website**.

**Comment by ShubhamTatvamasi** [comment]
> Also, I have a new framework suggestion. It's [astro](https://astro.build/)🚀. It uses .mdx instead of .md for content publication but customization it gives is out of the box. We can also use libraries like tailwind CSS in it. Just checkout. Let me know .

Is it simple to read and write?
Can we generate static web pages with this?

**Comment by Aditya-150** [comment]
I'm confirming about the best of the two within 2 days.

**Comment by ShubhamTatvamasi** [comment]
> I'm confirming about the best of the two within 2 days.

That will be great, Thanks.

**Comment by Aditya-150** [comment]
# Astro🚀 V/S Hugo🔥

This is a competitive analysis between 2 frameworks.
_Note: In this I have used basic boiler plate code that every framework gives._

GitHub Repo for Astro🚀: https://github.com/Aditya-150/hugo-test
GitHub Repo for Hugo🔥: https://github.com/Aditya-150/astro-blog

Both are hosted using Vercel. 

Count the emojis to decide the winner.
🚀 -> Astro
🔥 -> Hugo

| Framework |     SSG     |  Documentation  | Build Speed | Community <br/> Support | Markdown <br/> for Content Publication | Developer Experience | 3rd Party <br/> Library Integration | Real-time Data Loading | SEO | Performance <br/> (Lighthouse Score) | Deployment Ease |
|----------|:-------------:|:------:|:-------:|:-------:|:-------:|:-------:|:-------:|:-------:|:-------:|:-------:|:-------:|
| Hugo🔥 | <ul><li>[X] Checked</li></ul> | Good & Detailed | **4ms**🔥  | Larger🔥 & <br/> experienced | .md file supported + .json etc | Good | Can Work | May work | Good |  [Killer🔥](https://googlechrome.github.io/lighthouse/viewer/?psiurl=https%3A%2F%2Fhugo-test-hrq1f973k-aditya-150.vercel.app%2F&strategy=mobile&category=performance&category=accessibility&category=best-practices&category=seo&category=pwa&utm_source=lh-chrome-ext)| OK |
| Astro🚀 |    <ul><li>[X] Checked</li></ul> |   Good & Detailed | **407ms** for For normal template <br/> **546ms** for blog template  | Relatively New | .md + .mdx file supported🚀 | Better🚀 (fun for React developers) | Seamless🚀 | Works🚀 | Awesome🚀 | [Good](https://googlechrome.github.io/lighthouse/viewer/?psiurl=https%3A%2F%2Fastro-blog-pi-roan.vercel.app%2F&strategy=mobile&category=performance&category=accessibility&category=best-practices&category=seo&category=pwa&utm_source=lh-chrome-ext) | Better🚀|
| Winner⭐ |  _  | _ | 🔥 | 🔥 | 🚀 | 🚀 | 🚀 | 🚀 | 🚀 | 🔥 | 🚀|    

Some more online resources found:
- https://onebite.dev/hugo-vs-astro-which-static-site-generator-to-choose-in-2023/
- https://www.reddit.com/r/gohugo/comments/zjwplq/thoughts_on_astro_vs_hugo/?rdt=37655

(Based on data available on internet, some tests and our requirements)

_PS: I wasn't able to host Hugo site on GitHub pages due to some workflow issues for which I had to delete that repo🥲._

**Comment by ShubhamTatvamasi** [comment]
@Aditya-150 also check if these solutions can be deployed on GitHub pages.

**Comment by Aditya-150** [comment]
I will try again and let you know.

**Comment by Aditya-150** [comment]
Also if we host on Vercel we can still change the domain name by changing the DNS configuration.

**Comment by ShubhamTatvamasi** [comment]
> Also if we host on Vercel we can still change the domain name by changing the DNS configuration.

I guess Vercel is a paid service.

**Comment by Aditya-150** [comment]
Sorry, I thought it would work. Because it supports 100 deployments/ day.
So, I thought we won't exceed 3000 deployments on a month.

But for longetivity of this project this isn't right approach as this is an
open source project.

**Comment by Aditya-150** [comment]
@ShubhamTatvamasi just asking. Should I make any progress on the UI design part or wait for the content?

**Comment by ShubhamTatvamasi** [comment]
> @ShubhamTatvamasi just asking. Should I make any progress on the UI design part or wait for the content?

Hi @Aditya-150, please feel free to progress on the UI design part, we are still in the process of gathering all the content.

**Comment by Aditya-150** [comment]
Ok I'll be using content currently given in the website with some basic improvisations. Is that fine ? @ShubhamTatvamasi

**Comment by ShubhamTatvamasi** [comment]
> Ok I'll be using content currently given in the website with some basic improvisations. Is that fine ? @ShubhamTatvamasi

Sure, that would be great. Thanks @Aditya-150 😄

**Comment by hanshal101** [comment]
Hey everyone While reading the whole conversation, I understood that we are building the site on Hugo or Astro. My question is why are we not using Javascript frameworks like ReactJs or NextJs.

Also, I'm interested in developing the website.

**Comment by Aditya-150** [comment]
Hello @hanshal101. Nice to have you.
Thanks for going through the whole conversation. Also Thanks for showing so much interest in developing the website.

Coming to your question, 
We aren't actually building our website with `Hugo` or `Asto`. We are using **[docusaurus](https://docusaurus.io/) which is open source framework by Meta** for building our website which is a content driven framework too(_Refer to the docusaurus official documentation for understanding linked above👆_). Actually you are correct we could have used NEXTJS as it is a more dynamic framework and quite popular. We are using `docusaurus` as it is just a preference. Hope you understand.

Actually, currently we are at quite early stage of website. The first version is also yet to be released. There are a lot of thing to be done actually like documentation etc. So, you might need to wait for some time in order to start contributing. Thanks for showing so much interest. You can follow official documentation of [docusaurus](https://docusaurus.io/) and get to know about [kubearmor](https://docs.kubearmor.io/kubearmor/) till then.

PS: Join the Slack Channel if you want to be part of this journey. 👉[Join Slack](https://join.slack.com/t/kubearmor/shared_invite/zt-1ltmqdbc6-rSHw~LM6MesZZasmP2hAcA/)

**Comment by hanshal101** [comment]
Thanks, @Aditya-150 for such a nice explanation. I'm currently learning ReactJs for frontend development, so I was quite interested in building the KubeArmor website. I truly understand the current requirements, and you are right, it's quite challenging to maintain a NextJS or ReactJS website in the early stages.

I will definitely look up the Docusaurus documentation and learn more about this framework.
